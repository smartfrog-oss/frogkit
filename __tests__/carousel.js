import { mount, shallow } from 'vue-test-utils'
import Carousel from '@/components/carousel'

const propsData = {
  slides: Array.from({ length: 3 }).map((_, i) => ({file: `https://unsplash.it/320?random&${i}`, description: 'smartfrog cam'})),
  loop: true,
  active: 1  
}

describe('Carousel component', () => {
  it('Should render component and match snapshot', () => {
    const wrapper = mount(Carousel, { propsData })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Should move to the right slide without looping', () => {
    const cmp = shallow(Carousel, { propsData: { ...propsData, loop: false } })
    expect(cmp.vm.selected).toBe(1)
    cmp.vm.moveTo(2)
    expect(cmp.vm.selected).toBe(2)
    cmp.vm.moveTo(propsData.slides.length)
    expect(cmp.vm.selected).toBe(2)
    cmp.vm.moveTo(-1)
    expect(cmp.vm.selected).toBe(2)
  })

  it('Should move to the right slide when looping', () => {
    const cmp = shallow(Carousel, { propsData })
    expect(cmp.vm.selected).toBe(propsData.active)
    cmp.vm.moveTo(2)
    expect(cmp.vm.selected).toBe(2)
    cmp.vm.moveTo(propsData.slides.length)
    expect(cmp.vm.selected).toBe(0)
    cmp.vm.moveTo(-1)
    expect(cmp.vm.selected).toBe(propsData.slides.length - 1)
  })

  it('Should provide the right styles', () => {
    const cmp = shallow(Carousel, { propsData })
    cmp.vm.moveTo(1)
    expect(cmp.vm.styles).toEqual({
      transform: 'translateX(-100%)',
      transition: 'transform 300ms ease 0s',
    })
    cmp.vm.translate(1.5)
    expect(cmp.vm.styles).toEqual({
      transform: 'translateX(-150%)',
      transition: 'transform 0s linear 0s',
    })
  })
})
