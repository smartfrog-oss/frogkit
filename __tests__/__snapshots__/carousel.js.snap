// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component Should render component and match snapshot 1`] = `
<section class="fk-carousel">
  <swipeable>
    <div class="fk-carousel__track"><img src="https://unsplash.it/320?random&amp;0" alt="smartfrog cam" draggable="false" class="fk-carousel__slide"><img src="https://unsplash.it/320?random&amp;1" alt="smartfrog cam" draggable="false" class="fk-carousel__slide"><img src="https://unsplash.it/320?random&amp;2" alt="smartfrog cam" draggable="false" class="fk-carousel__slide"></div>
  </swipeable>
  <div class="fk-carousel__indicators"><label class="fk-carousel__indicator"></label><label class="fk-carousel__indicator fk-carousel__indicator--active"></label><label class="fk-carousel__indicator"></label></div>
  <flex justify="space-evenly" class="fk-carousel__previews"><img src="https://unsplash.it/320?random&amp;0" alt="smartfrog cam" draggable="false" class="fk-carousel__preview"><img src="https://unsplash.it/320?random&amp;1" alt="smartfrog cam" draggable="false" class="fk-carousel__preview fk-carousel__preview--active"><img src="https://unsplash.it/320?random&amp;2" alt="smartfrog cam" draggable="false" class="fk-carousel__preview"></flex>
</section>
`;
